auth-service/src/test/java/dev/rckft/authservice/controllers/AuthControllerIntegrationTest.java

```diff
@@ -0,0 +1,240 @@
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.web.reactive.server.WebTestClient;
+
+import java.util.Date;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@AutoConfigureWebTestClient
+class AuthControllerIntegrationTest {
+    private static final String EXISTING_TEST_USERNAME = "EXISTING_TEST_USERNAME";
+    private static final String EXISTING_TEST_PASSWORD = "EXISTING_TEST_PASSWORD";
+
+    private static final String NEW_TEST_USERNAME = "NEW_TEST_USERNAME";
+    private static final String NEW_TEST_PASSWORD = "NEW_TEST_PASSWORD";
+
+    @Autowired
+    private WebTestClient webTestClient;
+
+    @Autowired
+    private UserRepository userRepository;
+
+    @Autowired
+    private RevokedTokensRepository revokedTokensRepository;
+
+    @Autowired
+    private PasswordEncoder passwordEncoder;
+
+    @Autowired
+    private UserRegistrationService userRegistrationService;
+
+    @Autowired
+    private RevokedTokensService revokedTokensService;
+
+    @Autowired
+    private JwtUtil jwtUtil;
+
+    @Autowired
+    private JwtTestUtil jwtTestUtil;
+
+    @BeforeEach
+    void setUp() {
+        UserRegisterRequest request = new UserRegisterRequest(EXISTING_TEST_USERNAME, EXISTING_TEST_PASSWORD);
+        userRegistrationService.register(request);
+    }
+
+    @AfterEach
+    void tearDown() {
+        userRepository.deleteAll();
+        revokedTokensRepository.deleteAll();
+    }
+
+    @Test
+    void shouldRegisterUser() {
+        //given
+        UserRegisterRequest request = new UserRegisterRequest(NEW_TEST_USERNAME, NEW_TEST_PASSWORD);
+
+        //when then
+        webTestClient.post()
+                .uri("/api/auth/register")
+                .bodyValue(request)
+                .exchange()
+                .expectStatus().isCreated();
+
+        Optional<User> user = userRepository.findByUsername(NEW_TEST_USERNAME);
+
+        assertTrue(user.isPresent());
+        assertEquals(NEW_TEST_USERNAME, user.get().getUsername());
+        assertTrue(passwordEncoder.matches(NEW_TEST_PASSWORD, user.get().getPassword()));
+    }
+
+    @Test
+    void shouldNotRegisterUser_whenUserWithGivenUsernameExists() {
+        //given
+        UserRegisterRequest request = new UserRegisterRequest(EXISTING_TEST_USERNAME, EXISTING_TEST_PASSWORD);
+
+        //when then
+        webTestClient.post()
+                .uri("/api/auth/register")
+                .bodyValue(request)
+                .exchange()
+                .expectStatus().isBadRequest();
+
+        assertEquals(1, userRepository.findAll().size());
+    }
+
+    @Test
+    void shouldReturnJwtAccessAndRefreshTokens_whenLoggedIn() {
+        //given
+        AuthRequest authRequest = new AuthRequest(EXISTING_TEST_USERNAME, EXISTING_TEST_PASSWORD);
+
+        //when
+        AuthTokens responseBody = webTestClient.post()
```

Comment:
3x powtarzasz wysłanie loginu, może warto wrzucić to do metody prywatnej i zwrócić "exchange"? Wtedy każda metoda sobie sprawdzi co będzie chciała